src/components/Header.tsx

```diff
@@ -0,0 +1,69 @@
+import cn from 'classnames';
+import { useContext, useEffect, useRef, useState } from 'react';
+import { DispatchContext, StateContext } from '../Store';
+import { Todo } from '../types/todo';
+
+export const Header = () => {
+  const { todos, editingTodo } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editingTodo === null) {
+      inputRef.current?.focus();
+    }
+  }, [editingTodo, todos]);
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(e.target.value);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (newTodoTitle.trim() === '') {
```

Comment:
```suggestion
    if (!newTodoTitle.trim()) {
```
nit: empty string is a falsy value